### PR TITLE
Add analytics integration

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Frontend changes for https://github.com/home-assistant/core/pull/48256

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-analytics.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  TemplateResult,
+} from "lit-element";
+import { HomeAssistant } from "../types";
+import "./ha-icon-button";
+import { haStyle } from "../resources/styles";
+import type { HaCheckbox } from "./ha-checkbox";
+import { isComponentLoaded } from "../common/config/is_component_loaded";
+import { Analytics } from "../data/analytics";
+
+const preferences = ["base", "diagnostics", "usage", "statistics"];
```
Comment (ludeeus) [inline src/components/ha-analytics.ts:19]:
```suggestion
const PREFERENCES = ["base", "diagnostics", "usage", "statistics"];
```

File: src/components/ha-analytics.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    }
+    return html`
+      <p>
+        ${this.hass.localize(
+          "ui.panel.config.core.section.core.analytics.instance_id",
+          "huuid",
+          this.analytics.huuid
+        )}
+      </p>
+      ${preferences.map((preference) =>
```
Comment (ludeeus) [inline src/components/ha-analytics.ts:39]:
```suggestion
      ${PREFERENCES.map((preference) =>
```

File: src/components/ha-analytics.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                    )}
+              </span>
+            </ha-settings-row>`
+          : ""
+      )}
+      <p>
+        ${this.hass.localize(
+          "ui.panel.config.core.section.core.analytics.documentation",
+          "link",
+          html`<a href="https://www.home-assistant.io/integrations/analytics">
```
Comment (ludeeus) [inline src/components/ha-analytics.ts:83]: Use docklink URL helper

Comment (bramkragten) [reply src/components/ha-analytics.ts:83]: It is nice to see you reviewing your own PR's. I'll just leave you to it 👋

File: hassio/src/system/hassio-supervisor-info.ts [hunk 4]
```diff
@@ -148,30 +149,32 @@ class HassioSupervisorInfo extends LitElement {
...
-                      ${this.supervisor.localize("common.learn_more")}
-                    </button>
-                  </div>
-                  <ha-switch
-                    haptic
-                    .checked=${this.supervisor.supervisor.diagnostics}
-                    @change=${this._toggleDiagnostics}
-                  ></ha-switch>
-                </ha-settings-row>`
+              ? !atLeastVersion(this.hass.config.version, 2021, 4)
```
Comment (bramkragten) [inline hassio/src/system/hassio-supervisor-info.ts:152]: Is this because of backend changes?

File: src/components/ha-analytics.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    return html`
+      <p>
+        ${this.hass.localize(
+          "ui.panel.config.core.section.core.analytics.instance_id",
+          "huuid",
+          this.analytics.huuid
+        )}
+      </p>
+      ${PREFERENCES.map((preference) =>
+        preference === "base" || this.analytics.preferences.includes("base")
```
Comment (bramkragten) [inline src/components/ha-analytics.ts:40]: Should we instead of not showing the other options when `base` is not checked, disable them?

Comment (ludeeus) [reply src/components/ha-analytics.ts:40]: sure :+1:

File: src/components/ha-analytics.ts [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+import { HomeAssistant } from "../types";
+import "./ha-icon-button";
```
Comment (bramkragten) [inline src/components/ha-analytics.ts:13]: Unused?

File: src/components/ha-analytics.ts [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+    if (checkbox.checked) {
+      if (preferences.includes(preference)) {
+        return;
+      }
+      preferences.push(preference);
+    } else {
+      preferences = preferences.filter((entry) => entry !== preference);
+    }
+
+    this.analytics = { ...this.analytics, preferences };
```
Comment (bramkragten) [inline src/components/ha-analytics.ts:105]: we should not change the data we get from the parent, but instead, fire an event with a copied new version

File: src/data/analytics.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import { HomeAssistant } from "../types";
+
+export interface Analytics {
+  preferences: string[];
```
Comment (bramkragten) [inline src/data/analytics.ts:4]: Can we make this type more specific?

File: src/onboarding/ha-onboarding.ts [hunk 9]
```diff
@@ -102,6 +107,15 @@ class HaOnboarding extends litLocalizeLiteMixin(HassElement) {
         ></onboarding-core-config>
       `;
     }
+    if (step.step === "analytics") {
+      return html`
+        <onboarding-analytics
+          .hass=${this.hass}
+          .onboardingLocalize=${this.localize}
```
Comment (bramkragten) [inline src/onboarding/ha-onboarding.ts:114]:
```suggestion
          .localize=${this.localize}
```

File: src/onboarding/onboarding-analytics.ts [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+  LitElement,
+  property,
+  query,
+  TemplateResult,
+} from "lit-element";
+import { fireEvent } from "../common/dom/fire_event";
+import { LocalizeFunc } from "../common/translations/localize";
+import "../components/ha-analytics";
+import type { HaAnalytics } from "../components/ha-analytics";
+import "../components/map/ha-location-editor";
```
Comment (bramkragten) [inline src/onboarding/onboarding-analytics.ts:17]:
```suggestion
```

File: src/onboarding/onboarding-analytics.ts [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+      <p>
+        ${this.onboardingLocalize(
+          "ui.panel.page-onboarding.analytics.intro",
+          "link",
+          html`<a href="https://analaytics.home-assistant.io" target="_blank"
+            >https://analaytics.home-assistant.io</a
+          >`
+        )}
+      </p>
+      <ha-analytics .hass=${this.hass} .analaytics=${this.analyticsDetails}>
```
Comment (bramkragten) [inline src/onboarding/onboarding-analytics.ts:53]:
```suggestion
      <ha-analytics .hass=${this.hass} .analytics=${this.analyticsDetails}>
```

File: src/onboarding/onboarding-analytics.ts [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+        this.hass,
+        this._analytics?.analytics.preferences.includes("base")
+          ? this._analytics?.analytics.preferences
+          : []
+      );
+
+      const result = await onboardAnalyticsStep(this.hass);
+      fireEvent(this, "onboarding-step", {
+        type: "analytics",
+        result,
```
Comment (bramkragten) [inline src/onboarding/onboarding-analytics.ts:89]: I think this is optional? Do we have to send a response we are not doing anything with?

File: src/onboarding/onboarding-analytics.ts [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+        this._save(ev);
+      }
+    });
+    this._load();
+  }
+
+  private async _save(ev) {
+    ev.preventDefault();
+    try {
+      await setAnalyticsPrefrences(
```
Comment (bramkragten) [inline src/onboarding/onboarding-analytics.ts:79]:
```suggestion
      await setAnalyticsPreferences(
```

File: src/panels/config/core/ha-config-analytics.ts [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+        )}
+      >
+        <div class="card-content">
+          ${this._error ? html`<div class="error">${this._error}</div>` : ""}
+          <p>
+            ${this.hass.localize(
+              "ui.panel.config.core.section.core.analytics.introduction",
+              "link",
+              html`<a
+                href="https://analaytics.home-assistant.io"
```
Comment (bramkragten) [inline src/panels/config/core/ha-config-analytics.ts:60]:
```suggestion
                href="https://analytics.home-assistant.io"
```

File: src/panels/config/core/ha-config-analytics.ts [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
+                target="_blank"
+                >https://analaytics.home-assistant.io</a
```
Comment (bramkragten) [inline src/panels/config/core/ha-config-analytics.ts:62]:
```suggestion
                >https://analytics.home-assistant.io</a
```

File: src/panels/config/core/ha-config-analytics.ts [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+              >`
+            )}
+          </p>
+          <ha-analytics
+            .hass=${this.hass}
+            .analytics=${this.analyticsDetails}
+          ></ha-analytics>
+        </div>
+        <div class="card-actions">
+          <mwc-button @click=${this._save} .disabled=${false}>
```
Comment (bramkragten) [inline src/panels/config/core/ha-config-analytics.ts:72]: ?
```suggestion
          <mwc-button @click=${this._save}>
```

Comment (ludeeus) [reply src/panels/config/core/ha-config-analytics.ts:72]: I don't know...

File: src/components/ha-analytics.ts [hunk 17]
```diff
@@ -91,7 +149,7 @@ export class HaAnalytics extends LitElement {
   private _handleRowCheckboxClick(ev: Event) {
     const checkbox = ev.currentTarget as HaCheckbox;
     const preference = (checkbox as any).preference;
-    let preferences = this.analytics.preferences;
+    let preferences = this.analytics.preferences || [];
```
Comment (bramkragten) [inline src/components/ha-analytics.ts:152]:
```suggestion
    let preferences = [...(this.analytics.preferences || [])];
```

File: src/components/ha-analytics.ts [hunk 18]
```diff
@@ -28,6 +36,7 @@ export class HaAnalytics extends LitElement {
     if (!this.analytics.huuid) {
       return html``;
     }
+
```
Comment (bramkragten) [inline src/components/ha-analytics.ts:39]: const enabled = this.analytics.preferences.includes("base");

File: src/components/ha-analytics.ts [hunk 19]
```diff
@@ -36,52 +45,101 @@ export class HaAnalytics extends LitElement {
...
+                    )
                   : this.hass.localize(
-                      `ui.panel.config.core.section.core.analytics.preference.${preference}.title`
+                      `ui.panel.config.core.section.core.analytics.preference.usage.title`
+                    )
+                : this.hass.localize(
+                    `ui.panel.config.core.section.core.analytics.preference.${preference}.title`
+                  )}
+              ${!this.analytics.preferences.includes("base")
+                ? html`<ha-help-tooltip
```
Comment (bramkragten) [inline src/components/ha-analytics.ts:93]: I would just make this a normal tooltip on the disabled element I think?

File: src/panels/config/core/ha-config-analytics.ts [hunk 20]
```diff
@@ -0,0 +1,132 @@
...
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @internalProperty() private _analyticsDetails?: Analytics;
+
+  @internalProperty() private _error?: string;
+
+  protected render(): TemplateResult {
+    if (
+      !isComponentLoaded(this.hass, "analytics") ||
+      !this.hass.user?.is_admin ||
```
Comment (ludeeus) [inline src/panels/config/core/ha-config-analytics.ts:37]:
```suggestion
      !this.hass.user?.is_owner ||
```
?

File: src/components/ha-analytics.ts (hunk 19 above)
Comment (ludeeus) [reply src/components/ha-analytics.ts:93]: This looks cooler :D

File: src/panels/config/core/ha-config-analytics.ts (hunk 20 above)
Comment (bramkragten) [reply src/panels/config/core/ha-config-analytics.ts:37]: 🤷‍♂️

File: src/translations/en.json [hunk 21]
```diff
@@ -974,6 +974,35 @@
...
+              },
+              "analytics": {
+                "header": "Analytics",
+                "introduction": "Share analytics from your instance. This data will be publiclly available at {link}",
+                "instance_id": "Instance ID: {huuid}",
+                "needs_base": "You need to enable base analytics for this option to be available",
+                "preference": {
+                  "base": {
+                    "title": "Basic analytics",
+                    "description": "This includes the Instance ID, the version and the installation type"
```
Comment (spacegaier) [inline src/translations/en.json:986]:
```suggestion
                    "description": "This includes the instance ID, the version and the installation type"
```

**Review by bramkragten — APPROVED** [review]

**Review by bramkragten — APPROVED** [review]

**Pull request merged**